plugins/attachment-resources/src/components/AttachmentRefInput.svelte

```diff
@@ -39,24 +43,61 @@
 
   const client = getClient()
   const query = createQuery()
+  const drafts = writable<Record<Ref<Doc>, Record<Ref<Attachment>, Attachment>>>()
+
   let attachments: Map<Ref<Attachment>, Attachment> = new Map<Ref<Attachment>, Attachment>()
   let originalAttachments: Set<Ref<Attachment>> = new Set<Ref<Attachment>>()
   const newAttachments: Set<Ref<Attachment>> = new Set<Ref<Attachment>>()
   const removedAttachments: Set<Attachment> = new Set<Attachment>()
 
   let refContainer: HTMLElement
 
-  $: objectId &&
-    query.query(
-      attachment.class.Attachment,
-      {
-        attachedTo: objectId
-      },
-      (res) => {
-        originalAttachments = new Set(res.map((p) => p._id))
-        attachments = new Map(res.map((p) => [p._id, p]))
+  $: objectId && getAttachments(objectId)
+
+  async function getAttachments (objectId: Ref<Doc>) {
+    if (shouldUseDraft) {
+      drafts.set(fetchMetadataLocalStorage(attachment.metadata.AttachmentDrafts) ?? {})
```

Comment:
Please cancel query in this case, since if component will be reused with different props it will work incorrectly.

Reply:
Done